src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -0,0 +1,381 @@
+
+      # Verify ppa_id should not be None.
+      self.assertIsNotNone(ppa_id, msg="PPA ID received from SAS UUT as result of "
+                                       "PPA Creation is None")
+    except AssertionError as err:
+      logging.error('There is an error while creating PPA:%s', err.message)
+      self.fail('There is an error while creating PPA:%s' % err.message)
+    else:
+      logging.info('TriggerPpaCreation is in progress')
+
+      # Triggers most recent PPA Creation Status immediately and checks for the status of activity
+      # every 10 seconds until it is completed. If the status is not changed within 2 hours
+      # it will throw an exception.
+      signal.signal(signal.SIGALRM,
+                    lambda signum, frame:
+                    (_ for _ in ()).throw(
+                        Exception('Most Recent PPA Creation Status Check Timeout')))
+
+      # Timeout after 2 hours if it's not completed.
+      signal.alarm(7200)
+
+      # Check the Status of most recent ppa creation every 10 seconds.
+      while not self._sas_admin.GetPpaCreationStatus()['completed']:
+        time.sleep(10)
+
+      # Additional check to ensure whether PPA Creation status has error.
+      self.assertTrue(self._sas_admin.GetPpaCreationStatus()['completed'],
+                      msg='PPA creation status is not completed')
+      self.assertFalse(self._sas_admin.GetPpaCreationStatus()['withError'],
+                       msg='There was an error while creating PPA')
+      signal.alarm(0)
+
+      return ppa_id
+
+
+  def triggerCpasAndRetrievePpaZone(self, ppa_id, ssl_cert, ssl_key):
+    """ Triggers CPAS and Retrieves PPA Zone Record matches with specified ppa_id.
+
+    Performs CPAS and pulls FAD from SAS UUT. Retrieves the ZoneData Records from FAD,
+    checks that only one record is present and it matches the ppa_id.
+
+    Args:
+      ppa_id: String format of PPA ID.
+      ssl_cert: Path to SSL cert file to be used for pulling FAD record.
+      ssl_key: Path to SSL key file to be used for pulling FAD record.
+
+    Returns:
+      A PPA record of format of ZoneData Object.
+
+    """
+
+    # Notify the SAS UUT about the SAS Test Harness.
+    certificate_hash = getCertificateFingerprint(ssl_cert)
+    self._sas_admin.InjectPeerSas({'certificateHash': certificate_hash,
+                                   'url': SAS_TEST_HARNESS_URL})
+
+    # As SAS is reset at the beginning of the test, the FAD records should contain
+    # only one zone record containing the PPA that was generated. Hence the first
+    # zone record is retrieved and verified if it matches the PPA ID.
+    uut_fad = getFullActivityDumpSasUut(self._sas, self._sas_admin, ssl_cert, ssl_key)
+
+    try:
+
+      # Check if the retrieved FAD that has valid PPA zone record matches the PPA ID
+      # that was generated using the admin API.
+      uut_ppa_zone_data = uut_fad.getZoneRecords([lambda x: x['id'] == ppa_id])
```

Comment:
I'm not sure there's a requirement that the PPA ID here be the same as the one returned from the API. It's likely to be the same, but I don't think we would want to fail a SAS if they did something slightly different.

Instead, I think you can verify that there's only one PPA in the FAD and then use that one. If it's somehow the wrong PPA, the SAS will fail anyway at the check step.

Reply:
Agreed.   The test spec mentions to retrieve the PPA info using the PPA ID. However the   reasoning you mentioned makes sense and we have updated the code accordingly.